src/main/java/com/tterrag/k9/mappings/official/OfficialMapping.java

```diff
@@ -0,0 +1,122 @@
+
+    @Getter(onMethod = @__(@Override))
+    private final MappingType type;
+
+    private final Mapping owner;
+
+    private final String desc;
+
+    @Getter(onMethod = @__(@Override))
+    private final String original, name, memberClass;
+
+    @ToString.Exclude
+    private final transient Map<NameType, String> mappedOwner = new EnumMap<>(NameType.class), mappedDesc = new EnumMap<>(NameType.class);
+
+    @NonFinal
+    private String intermediate = null;
+
+    @Override
+    public String formatMessage(String mcver) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("\n");
+        builder.append("**MC " + mcver + ": " + (owner == null ? "" : owner.getName() + ".") + name + "**\n");
+        String intermediate = getIntermediate();
+        builder.append("__Name__: `" + original + (intermediate.isEmpty() ? "" : "` => `" + intermediate) + "` => `" + name + "`\n");
+
+        builder.append("__Side__: `" + side + "`");
+
+        if (getType() != MappingType.PARAM && !intermediate.isEmpty()) {
+            builder.append("\n__AT__: `public ").append(Strings.nullToEmpty(getOwner(NameType.INTERMEDIATE)).replace('/', '.'));
+            String atName = intermediate;
+            if (getType() == MappingType.CLASS) {
+                atName = atName.replace('/', '.');
+            } else {
+                // If this is a class, then getOwner() is empty meaning we shouldn't add another space
+                builder.append(' ');
+            }
+            builder.append(atName);
+            String desc = getDesc();
+            if (desc != null) {
+                builder.append(desc);
+            }
+            if (getType() != MappingType.CLASS) {
+                builder.append(" # ").append(getName() == null ? intermediate : getName());
+            }
+            builder.append("`");
+        }
+
+        if (getMemberClass() != null)
+            builder.append("\n__Type__: `").append(getMemberClass()).append("`");
+
+        return builder.toString();
+    }
+
+    OfficialMapping cloneWithSide(McpMapping.Side side) {
+        return new OfficialMapping(srgs, db, side, type, owner, desc, original, name, memberClass);
+    }
+
+    @Override
+    public @Nullable String getOwner() {
+        return getOwner(NameType.NAME);
+    }
+
+    @Override
+    public @Nullable String getOwner(NameType name) {
+        return mappedOwner.computeIfAbsent(name, t -> owner == null ? null : sigHelper.mapType(t, owner.getOriginal(), this, db).getInternalName());
+    }
+
+    private String mapType(NameType t, String type) {
+        return sigHelper.mapType(t, Type.getType(type), this, db).getDescriptor();
+    }
+
+    @Override
+    public String getDesc(NameType name) {
+        return mappedDesc.computeIfAbsent(name, t -> desc == null ? null : desc.contains("(") ? sigHelper.mapSignature(t, desc, this, db) : mapType(t, desc));
+    }
+
+    @Override
+    public String getIntermediate() {
```

Comment:
When first loading, it seems this method is called for every mapping for some reason and causes long load times for initially running a command relating to Official mappings. I'm not sure how to improve it or mitigate this issue without other issues.

Reply:
What you could do is extend AbstractMappingDatabase to expose a way to customize which NameTypes are valid for lookups, then exclude intermediate names. That way they would never even get entered into the table for lookups. The downside is it would prevent looking names up by srg. Up to you honestly.

Reply:
I've updated the loading mechanism by populating some maps in `SrgDatabase` that map original class names to the SrgMapping instance, and map owner class original names to all the children. Changing to this method vastly improved loading times (from 1-2 minutes, sometimes exceeding timeout length) to under 10 seconds. I also made it so this information is only stored if a boolean flag is enabled in the constructor of the database (with an overloaded constructor defauting to false), otherwise the maps are not populated.

This solution seems a lot more sound and reasonable, as I was having issues with the loading taking longer than a minute and being cut off by the timeout limit of 1 minute. By storing maps with this information, a lot of dupllicate info is weeded out as searching the table every single time is not efficient whatsoever. Marking as resolved.